### PR TITLE
CAS-169 - update fields validation on community creation

### PR DESCRIPTION
**Pull request by germanurrus**
Update on community fields: 
community name => required
community Description => not required
logo => not required, defaults to `react-blockies`
community Terms => not required      
category => required

### CODE REVIEW

File: frontend/packages/client/src/hooks/useCommunityDetails.js [hunk 1]
```diff
@@ -18,18 +18,14 @@ export default function useCommunityDetails(id) {
       const response = await fetch(url);
       const details = await checkResponse(response);
       // merging with mock data
-      const detailsMocked = {
-        logo: 'https://i.imgur.com/RMKXPCw.png',
-        name: 'Flow',
-        description:
-          "Flow's decentralized governance and node operations community.",
```
Comment (germanurrus) [inline frontend/packages/client/src/hooks/useCommunityDetails.js:25]: we don't use the logo from flow anymore as default

File: frontend/packages/client/src/hooks/useCommunity.js [hunk 2]
```diff
@@ -83,9 +83,9 @@ export default function useCommunity() {
         } = communityData;
 
         let communityLogo;
-        // not handling upload error: there's a default image
+        // check for logo upload
         // admins can edit later the image
-        if (logo.file) {
+        if (logo?.file) {
```
Comment (germanurrus) [inline frontend/packages/client/src/hooks/useCommunity.js:88]: check when creating community if logo was added or not

File: frontend/packages/client/src/components/CommunityCreate/StepOne.js [hunk 3]
```diff
@@ -107,10 +107,13 @@ export default function StepOne({
...
-      logo: (logo) => logo?.file && logo?.imageUrl,
+      communityDescription: (desc) =>
+        desc?.trim().length ? desc?.trim().length < 1000 : true,
+      logo: (logo) =>
+        logo !== undefined ? logo?.file && logo?.imageUrl : true,
       communityTerms: (termsUrl) =>
-        termsUrl?.length > 0 && urlPatternValidation(termsUrl),
+        termsUrl?.length > 0 ? urlPatternValidation(termsUrl) : true,
+      category: (cat) => cat?.length > 0,
     };
```
Comment (germanurrus) [inline frontend/packages/client/src/components/CommunityCreate/StepOne.js:117]: most of the important changes are here

File: frontend/packages/client/src/components/Community/CommunityCard.js [hunk 4]
```diff
@@ -55,15 +55,24 @@ const CommunityCard = ({ logo, name, description, id }) => {
...
+                  }}
+                />
+              ) : (
+                <Blockies
+                  seed={slug}
+                  size={10}
+                  scale={9.6}
+                  className="rounded-sm"
+                />
+              )}
```
Comment (germanurrus) [inline frontend/packages/client/src/components/Community/CommunityCard.js:75]: Screen Shot 2022-07-01 at 17 53 54

File: frontend/packages/client/src/pages/Community.js [hunk 5]
```diff
@@ -234,29 +242,45 @@ export default function Community() {
...
+                    width="85px"
+                  />
+                ) : (
+                  <Blockies
+                    seed={slug ?? id}
+                    size={10}
+                    scale={9.6}
+                    className="blockies"
+                  />
+                )}
```
Comment (germanurrus) [inline frontend/packages/client/src/pages/Community.js:283]: Screen Shot 2022-07-01 at 17 53 48

**Review by CJamesKeller — APPROVED** [review]

**Review by dbslone — APPROVED** [review]

**Pull request merged**